src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -761,12 +766,28 @@ void CCalcEngine::ResolveHighestPrecedenceOperation()
         {
             m_currentVal = m_holdVal;
             DisplayNum(); // to update the m_numberString
-            m_HistoryCollector.AddBinOpToHistory(m_nOpCode, false);
-            m_HistoryCollector.AddOpndToHistory(m_numberString, m_currentVal); // Adding the repeated last op to history
+            if (IsBinOpCode(m_nOpCode))
+            {
+                m_HistoryCollector.AddBinOpToHistory(m_nOpCode, false);
+                m_HistoryCollector.AddOpndToHistory(m_numberString, m_currentVal); // Adding the repeated last op to history
+            }
+            else if (IsUnaryOpCode(m_nOpCode) && m_nOpCode != IDC_PERCENT)
+            {
+                m_HistoryCollector.AddUnaryOpToHistory(m_nOpCode, m_bInv, m_angletype);
```

Comment:
Sticking with this code from the previous comment...

https://github.com/microsoft/calculator/blob/80912a33d0a4a853ebb43e312999a9fd6197d32e/src/CalcManager/CEngine/scicomm.cpp#L294-L304

Can you clarify why you don't feel like we need to call `m_HistoryCollector.AddOpndToHistory(m_numberString, m_currentVal);` here the way we're doing for the binary operators?

Reply:
Of course! The context of this code and the one mentioned are different.

**Line 298:** 
**Context:** the user just entered a unitary operator
When the user enters a unitary operator, there is a high chance that the number he has previously typed (displayed in the CalculationResult control) has not been added to the history, so we must first check if we should add the number before adding the unitary operator.

**Line 774** 
**Context:** the user typed more than 1 time on <kbd>=</kbd>
Users has not typed a number since the last <kbd>=</kbd> (this code is only executed when nothing has been typed since the last one, so we are 100% sure there is no operands to add to the history. The only thing we must do is to add the unit operator.